clippy_lints/src/missing_asserts_for_indexing.rs

```diff
@@ -134,18 +135,30 @@ fn assert_len_expr<'hir>(
     cx: &LateContext<'_>,
     expr: &'hir Expr<'hir>,
 ) -> Option<(LengthComparison, usize, &'hir Expr<'hir>)> {
-    if let Some(higher::If { cond, then, .. }) = higher::If::hir(expr)
+    let (cmp, asserted_len, slice_len) = if let Some(higher::If { cond, then, .. }) = higher::If::hir(expr)
         && let ExprKind::Unary(UnOp::Not, condition) = &cond.kind
         && let ExprKind::Binary(bin_op, left, right) = &condition.kind
-
-        && let Some((cmp, asserted_len, slice_len)) = len_comparison(*bin_op, left, right)
-        && let ExprKind::MethodCall(method, recv, [], _) = &slice_len.kind
-        && cx.typeck_results().expr_ty_adjusted(recv).peel_refs().is_slice()
-        && method.ident.name == sym::len
-
         // check if `then` block has a never type expression
         && let ExprKind::Block(Block { expr: Some(then_expr), .. }, _) = then.kind
         && cx.typeck_results().expr_ty(then_expr).is_never()
+    {
+        len_comparison(bin_op.node, left, right)?
+    } else if let Some((macro_call, bin_op)) =
+        first_node_macro_backtrace(cx, expr).find_map(|macro_call| match cx.tcx.item_name(macro_call.def_id).as_str() {
+            "assert_eq" => Some((macro_call, BinOpKind::Eq)),
+            "assert_ne" => Some((macro_call, BinOpKind::Ne)),
```

Comment:
I think you can use the symbols directly here (for example https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/symbol/sym/constant.assert_ne_macro.html) instead of matching on strings.

Reply:
Good point. Updated.